crates/apub/src/objects/post.rs

```diff
@@ -225,6 +230,23 @@ impl Object for ApubPost {
       None
     };
 
+    // If NSFW is not allowed, reject NSFW posts and delete existing
+    // posts that get updated to be NSFW
+    let block_for_nsfw = check_nsfw_allowed(page.sensitive, local_site.as_ref());
+    if block_for_nsfw.is_err() {
+      let post = ApubPost::read_from_id(page.id.inner().clone(), context).await?;
+      if let Some(post) = post {
+        if let Some(url) = &post.url {
+          purge_image_from_pictrs(url, context).await.ok();
+        }
+        if let Some(thumbnail_url) = &post.thumbnail_url {
+          purge_image_from_pictrs(thumbnail_url, context).await.ok();
+        }
+        ApubPost::delete(post, context).await?;
```

Comment:
You can make a function `purge_post()` in api_common/src/utils, as the same logic is also used in `purge_image_posts_for_person` and `purge_image_posts_for_community`.